### ISSUE TITLE
grains['oscodename'] incorrect for Debian testing.

### ISSUE THREAD

**Issue by optix2000** [saltstack/salt#30123]
`grains['oscodename']` seems to be parsed incorrectly from `/etc/os-release` on Debian testing, due to the slash in the codename.

For Debian-based distros (or just Debian), the codename should probably be pulled from `/etc/debian_version` since it's much easier to parse without error.

```
optix2000@inazuma ~ % cat /etc/debian_version
stretch/sid
optix2000@inazuma ~ % cat /etc/os-release
PRETTY_NAME="Debian GNU/Linux stretch/sid"
NAME="Debian GNU/Linux"
ID=debian
HOME_URL="https://www.debian.org/"
SUPPORT_URL="https://www.debian.org/support"
BUG_REPORT_URL="https://bugs.debian.org/"
optix2000@inazuma ~ % sudo salt-call grains.get oscodename
local:
    sid
optix2000@inazuma ~ % salt-call --versions-report
Salt Version:
           Salt: 2015.8.3

Dependency Versions:
         Jinja2: 2.8
       M2Crypto: Not Installed
           Mako: 1.0.3
         PyYAML: 3.11
          PyZMQ: 14.4.0
         Python: 2.7.11 (default, Dec  9 2015, 00:29:25)
           RAET: Not Installed
        Tornado: 4.2.1
            ZMQ: 4.0.5
           cffi: Not Installed
       cherrypy: Not Installed
       dateutil: 2.4.2
          gitdb: Not Installed
      gitpython: Not Installed
          ioflo: Not Installed
        libnacl: Not Installed
   msgpack-pure: Not Installed
 msgpack-python: 0.4.6
   mysql-python: 1.3.7
      pycparser: Not Installed
       pycrypto: 2.6.1
         pygit2: Not Installed
   python-gnupg: Not Installed
          smmap: Not Installed
        timelib: Not Installed

System Versions:
           dist: debian stretch/sid
        machine: x86_64
        release: 4.3.0-1-amd64
         system: debian stretch/sid
```

**Comment by cachedout** [comment]
Sounds good. Obviously, we'll want to proceed conservatively here. Lots of people use this grain for targeting but handling the case of a slash seems sensible.

Did you want to work up a PR for this one, @optix2000 ?

**Comment by optix2000** [comment]
Yeah, I'll work up a PR for this. Wanted to see if anyone had any concerns before I started touching it.

**Comment by optix2000** [comment]
Looks like this was a bug in what `lsb-release` returned to salt. This is fixed upstream in `lsb-release 9.20160110`.

Closing, as not an issue in salt.
